src/main/java/org/zalando/nakadi/validation/SchemaEvolutionService.java

```diff
@@ -97,6 +102,12 @@ public EventType evolve(final EventType original, final EventTypeBase eventType)
 
         final Version.Level changeLevel = semanticOfChange(changes, original.getCompatibilityMode());
 
+        if (changeLevel == NO_CHANGES && !original.getSchema().getSchema().equals(eventType.getSchema().getSchema())) {
+            LOG.error("undetected schema changes from {} to {}", original.getSchema().getSchema(),
+                    eventType.getSchema().getSchema());
+            throw new MyNakadiRuntimeException1("undetected schema changes");
```

Comment:
@v-stepanov do you think that if I created a specific exception type for this it would be better? Like `UnexpectedSchemaChangeException`?

About the ticket, the way to solve this is not by improving the `diff` operation, but to avoid users from defining weird properties like `example`. Since the ticket tile was about consistency between the information in tables event_type and event_type_schema, I just added enough code to avoid this inconsistency from happening.

But I'll keep coding and implement additional validations that will keep users from defining such weird schemas with invalid properties.